Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -561,6 +561,7 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
                       , toNubListR (cxxSources libBi)
                       , toNubListR (cmmSources libBi)
                       , toNubListR (asmSources libBi)
+                      , toNubListR (jsSources  libBi)
```

Comment:
This is adding JS to `cLikeSources`. Is "C-like" a misnomer or was the original intention to exclude JS? Change of plans? I don't see any "non-C-like sources" list in the codebase that we could add JS source to, instead, so this must be the right place. 

Perhaps the C-like vs non-C-like distinction is a left over from some special cases for GHCJS? Is JS C-like enough when considered by GHC JS backend, but not C-like when fed to GHCJS?

Reply:
Yes for the JS backend we've reworked the way JS sources are treated. They are now C-like because we've added support for a "compilation" from `.js` to `.o` files (basically a file renaming with a header added). It makes them less special for tools like Hadrian and cabal.

(in the GHCJS code path (`Distribution.Simple.GHCJS`) these files were directly passed as link options)

Reply:
Maybe this deserves a `NOTE JS Sources are C-Like` or similar.

Reply:
@Profpatsch: GHC note or cabal note? Or one of those and hyperlink from the other place?

Reply:
I've added a comment explaining that JS sources are C-like in the JS backend.